sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ControllingPermissions.CSharp9.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Security.Permissions;
+using System.Security.Principal;
+using System.Threading;
+
+// Method declarations that accept IIdentity or IPrincipal
+void AcceptIdentity1(MyIdentity identity) { } // Compliant - FN
+void AcceptIdentity2(IIdentity identity) { } // Compliant - FN
+void AcceptPrincipal1(MyPrincipal principal) { } // Compliant - FN
+void AcceptPrincipal2(IPrincipal principal) { } // Compliant - FN
```

Comment:
Only in case we need this test:
```suggestion
void AcceptIdentity1(MyIdentity identity) { } // FN
void AcceptIdentity2(IIdentity identity) { } // FN
void AcceptPrincipal1(MyPrincipal principal) { } // FN
void AcceptPrincipal2(IPrincipal principal) { } // FN
```